main/reveal/index.ts

```diff
@@ -0,0 +1,115 @@
+// Reveal details about pending transactions
+
+import log from 'electron-log'
+// @ts-ignore
+import EthereumProvider from 'ethereum-provider'
+import proxyConnection from '../provider/proxy'
+import nebulaApi from '../nebula'
+
+import Erc20Contract from '../contracts/erc20'
+import { decodeContractCall } from '../contracts'
+
+const nebula = nebulaApi()
+const provider = new EthereumProvider(proxyConnection)
+
+// TODO: Discuss the need to set chain for the proxy connection
+provider.setChain('0x1')
+
+type EntityType = 'external' | 'contract' | 'unknown'
+
+async function resolveEntityType (address: string, chainId: string): Promise<EntityType> {
+  if (!address || !chainId) return 'unknown'
+  try {
+    const payload: JSONRPCRequestPayload = {
+      method: 'eth_getCode',
+      params: [address, 'latest'],
+      jsonrpc: '2.0',
+      id: 1,
+      chainId // TODO: Verify this overrides setChain
+    }      
+    const code = (await provider.request(payload) || [])
+    const type = (code === '0x' || code === '0x0') ? 'external' : 'contract'
+    return type
+  } catch (e) {
+    log.error(e)
+    return 'unknown'
+  }
+}
+
+async function resolveEnsName (address: string): Promise<string> {
+  try {
+    const ensName: string = (await nebula.ens.reverseLookup([address]) || [])[0]
```

Comment:
I think this reverseLookup returns `['']` when it fails so you might not need the default:
https://github.com/frame-labs/nebula/blob/f863d1e41bd0248c3cd42f8cced8baa12af6dd13/src/eth/ens/index.ts#L162